src/zino/api/legacy.py

```diff
@@ -91,13 +92,18 @@ def connection_lost(self, exc: Optional[Exception]) -> None:
             self.notification_channel.goodbye()
 
     def data_received(self, data):
-        try:
-            message = data.decode().rstrip("\r\n")
-        except UnicodeDecodeError:
-            _logger.error("Received garbage server input from %s: %r", self.peer_name, data)
-            self.transport.close()
-            return
-        _logger.debug("Data received from %s: %r", self.peer_name, message)
+        self._input_buffer.extend(data)
+        while b"\n" in self._input_buffer:
+            line, self._input_buffer = self._input_buffer.split(b"\n", 1)
```

Comment:
Telnet sends `\r\n` but curitz may send `\n`.

Reply:
Indeed, which is why the code splits on `\n` and strips `\r`, which should support both cases.